src/topology/continuous_function/compact.lean

```diff
@@ -428,3 +428,86 @@ end
 end weierstrass
 
 end continuous_map
+
+/-!
+### Star structures
+
+In this section, if `β` is a normed ⋆-group, then so is the space of
+continuous functions from `α` to `β`, by using the star operation pointwise.
+
+If `𝕜` is normed field and a ⋆-ring over which `β` is a normed algebra and a
+star module, then the space of continuous functions from `α` to `β`
+is a star module.
+
+If `β` is a ⋆-ring in addition to being a normed ⋆-group, then `C(α, β)`
+inherits a ⋆-ring structure.
+
+Furthermore, if `α` is compact and `β` is a C⋆-ring, then `C(α, β)` is a C⋆-ring.  -/
+
+section continuous_star
+
+variables {α : Type*} {β : Type*}
+variables [topological_space α] [topological_space β] [add_monoid β] [has_continuous_add β]
+  [star_add_monoid β] [has_continuous_star β]
+
+instance : star_add_monoid C(α, β) :=
+{ star            := λ f, star_continuous_map.comp f,
+  star_involutive := λ f, by { ext, exact star_star _ },
+  star_add        := λ f g, by { ext, exact star_add _ _ } }
+
+@[simp] lemma coe_star (f : C(α, β)) : ⇑(star f) = star f := rfl
+
+@[simp] lemma star_apply (f : C(α, β)) (x : α) : star f x = star (f x) := rfl
+
+end continuous_star
```

Comment:
I'd probably put this in `topology/continuous_function/algebra`, since it's a pretty trivial definition

Reply:
Fair enough -- done!